Sources/Amplitude/Plugins/iOS/IOSLifecycleMonitor.swift

```diff
@@ -67,22 +66,35 @@
 
         func applicationWillEnterForeground(notification: NSNotification) {
             let timestamp = Int64(NSDate().timeIntervalSince1970 * 1000)
+
```

Comment:
Can this be improved as the following? or even use a simple if statement instead?

```
            let fromBackground: Bool = false;
            if let sharedApplication = application {
                switch sharedApplication.applicationState {
                case .background:
                    fromBackground = true
                }
            } 
```

Reply:
So yes, the code could be simplified to something like:
```
var fromBackground: Bool = false;
if let sharedApplication = application, sharedApplication.applicationState == .background {
  fromBackground = true
}
```

However, the current structure has a few better properties:
- fromBackground is a `let` vs a `var` which means the compiler will enforce that it defined exactly once on every code path and never modified.
- Using a switch statement with all cases explicitly defined will warn us if any future cases are added to UIApplicationState vs silently breaking. (Also a good reason to avoid using the default case for non-frozen enums you don't own.)